symphony-bdk-legacy/symphony-api-client-java/src/test/java/services/AbstractDatafeedEventsServiceTest.java

```diff
@@ -0,0 +1,248 @@
+        newSize = service.getConnectionListeners().size();
+        assertEquals(initialSize - 2, newSize);
+    }
+
+    @Test
+    public void removeElementsListenersTest() {
+        MyAbstractDatafeedEventsService service = new MyAbstractDatafeedEventsService(symBotClient);
+        ElementsListener elementsListener1 = this.createElementsListener();
+        ElementsListener elementsListener2 = this.createElementsListener();
+        ElementsListener elementsListener3 = this.createElementsListener();
+        service.addListeners(elementsListener1, elementsListener2, elementsListener3);
+        int initialSize = service.getElementsListener().size();
+        int newSize;
+
+        service.removeListeners(elementsListener1);
+        newSize = service.getElementsListener().size();
+        assertEquals(initialSize - 1, newSize);
+
+        service.removeListeners(elementsListener3);
+        newSize = service.getElementsListener().size();
+        assertEquals(initialSize - 2, newSize);
+    }
+
+    @Test
+    public void removeIMListenersTest() {
+        MyAbstractDatafeedEventsService service = new MyAbstractDatafeedEventsService(symBotClient);
+        IMListener imListener1 = this.createIMListener();
+        IMListener imListener2 = this.createIMListener();
+        IMListener imListener3 = this.createIMListener();
+        service.addListeners(imListener1, imListener2, imListener3);
+        int initialSize = service.getIMListener().size();
+        int newSize;
+
+        service.removeListeners(imListener1);
+        newSize = service.getIMListener().size();
+        assertEquals(initialSize - 1, newSize);
+
+        service.removeListeners(imListener3);
+        newSize = service.getIMListener().size();
+        assertEquals(initialSize - 2, newSize);
+    }
+
+    // Anonymous class creation
+    private RoomListener createRoomListener() {
+        return new RoomListener() {
+            @Override
+            public void onRoomMessage(InboundMessage message) {}
+            @Override
+            public void onRoomCreated(RoomCreated roomCreated) {}
+            @Override
+            public void onRoomDeactivated(RoomDeactivated roomDeactivated) {}
+            @Override
+            public void onRoomMemberDemotedFromOwner(RoomMemberDemotedFromOwner roomMemberDemotedFromOwner) {}
+            @Override
+            public void onRoomMemberPromotedToOwner(RoomMemberPromotedToOwner roomMemberPromotedToOwner) {}
+            @Override
+            public void onRoomReactivated(Stream stream) {}
+            @Override
+            public void onRoomUpdated(RoomUpdated roomUpdated) {}
+            @Override
+            public void onUserJoinedRoom(UserJoinedRoom userJoinedRoom) {}
+            @Override
+            public void onUserLeftRoom(UserLeftRoom userLeftRoom) {}
+        };
+    }
+
+    private ConnectionListener createConnectionListener() {
+        return new ConnectionListener() {
```

Comment:
Same comment as above, could we use `mock(ConnectionListener.class)` to generate new instances?